File: src/main/java/Models/Application.java
package Models;

import java.time.LocalDate;
import java.time.LocalDateTime;

public class Application {

    private int applicationId;
    private LocalDate applicationDate;
    private String coverLetter;
    private String currentStatus;
    private String resumePath;
    private LocalDateTime lastUpdateDate;
    private int user_id;
    private int jobOfferId;

    // New realistic fields
    private double expectedSalary;
    private LocalDate availabilityDate;
    private String phone;
    private String email;
    private int experienceYears;
    private String portfolioUrl;
    private Double score;      // recruiter evaluation (nullable)
    private String reviewNote; // recruiter note

    public Application() {}

    // Constructor for adding new application (candidate)
    public Application(LocalDate applicationDate, String coverLetter, String resumePath,
                       int user_id, int jobOfferId,
                       double expectedSalary, LocalDate availabilityDate,
                       String phone, String email, int experienceYears, String portfolioUrl) {

        this.applicationDate = applicationDate;
        this.coverLetter = coverLetter;
        this.resumePath = resumePath;
        this.user_id = user_id;
        this.jobOfferId = jobOfferId;
        this.expectedSalary = expectedSalary;
        this.availabilityDate = availabilityDate;
        this.phone = phone;
        this.email = email;
        this.experienceYears = experienceYears;
        this.portfolioUrl = portfolioUrl;
    }

    // Constructor for reading from DB
    public Application(int applicationId, LocalDate applicationDate,
                       String coverLetter, String currentStatus,
                       String resumePath, LocalDateTime lastUpdateDate,
                       int user_id, int jobOfferId,
                       double expectedSalary, LocalDate availabilityDate,
                       String phone, String email, int experienceYears,
                       String portfolioUrl, Double score, String reviewNote) {

        this.applicationId = applicationId;
        this.applicationDate = applicationDate;
        this.coverLetter = coverLetter;
        this.currentStatus = currentStatus;
        this.resumePath = resumePath;
        this.lastUpdateDate = lastUpdateDate;
        this.user_id = user_id;
        this.jobOfferId = jobOfferId;
        this.expectedSalary = expectedSalary;
        this.availabilityDate = availabilityDate;
        this.phone = phone;
        this.email = email;
        this.experienceYears = experienceYears;
        this.portfolioUrl = portfolioUrl;
        this.score = score;
        this.reviewNote = reviewNote;
    }

    // ===== Getters & Setters =====

    public int getApplicationId() { return applicationId; }
    public void setApplicationId(int applicationId) { this.applicationId = applicationId; }

    public LocalDate getApplicationDate() { return applicationDate; }
    public void setApplicationDate(LocalDate applicationDate) { this.applicationDate = applicationDate; }

    public String getCoverLetter() { return coverLetter; }
    public void setCoverLetter(String coverLetter) { this.coverLetter = coverLetter; }

    public String getCurrentStatus() { return currentStatus; }
    public void setCurrentStatus(String currentStatus) { this.currentStatus = currentStatus; }

    public String getResumePath() { return resumePath; }
    public void setResumePath(String resumePath) { this.resumePath = resumePath; }

    public LocalDateTime getLastUpdateDate() { return lastUpdateDate; }
    public void setLastUpdateDate(LocalDateTime lastUpdateDate) { this.lastUpdateDate = lastUpdateDate; }

    public int getuser_id() { return user_id; }
    public void setUser_id(int user_id) { this.user_id = user_id; }

    public int getJobOfferId() { return jobOfferId; }
    public void setJobOfferId(int jobOfferId) { this.jobOfferId = jobOfferId; }

    public double getExpectedSalary() { return expectedSalary; }
    public void setExpectedSalary(double expectedSalary) { this.expectedSalary = expectedSalary; }

    public LocalDate getAvailabilityDate() { return availabilityDate; }
    public void setAvailabilityDate(LocalDate availabilityDate) { this.availabilityDate = availabilityDate; }

    public String getPhone() { return phone; }
    public void setPhone(String phone) { this.phone = phone; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public int getExperienceYears() { return experienceYears; }
    public void setExperienceYears(int experienceYears) { this.experienceYears = experienceYears; }

    public String getPortfolioUrl() { return portfolioUrl; }
    public void setPortfolioUrl(String portfolioUrl) { this.portfolioUrl = portfolioUrl; }

    public Double getScore() { return score; }
    public void setScore(Double score) { this.score = score; }

    public String getReviewNote() { return reviewNote; }
    public void setReviewNote(String reviewNote) { this.reviewNote = reviewNote; }
}

File: src/Entity/Application.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;

use App\Repository\ApplicationRepository;

#[ORM\Entity(repositoryClass: ApplicationRepository::class)]
#[ORM\Table(name: 'application')]
class Application
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $applicationId = null;

    public function getApplicationId(): ?int
    {
        return $this->applicationId;
    }

    public function setApplicationId(int $applicationId): self
    {
        $this->applicationId = $applicationId;
        return $this;
    }

    #[ORM\Column(type: 'date', nullable: true)]
    private ?\DateTimeInterface $applicationDate = null;

    public function getApplicationDate(): ?\DateTimeInterface
    {
        return $this->applicationDate;
    }

    public function setApplicationDate(?\DateTimeInterface $applicationDate): self
    {
        $this->applicationDate = $applicationDate;
        return $this;
    }

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $coverLetter = null;

    public function getCoverLetter(): ?string
    {
        return $this->coverLetter;
    }

    public function setCoverLetter(?string $coverLetter): self
    {
        $this->coverLetter = $coverLetter;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: true)]
    private ?string $currentStatus = null;

    public function getCurrentStatus(): ?string
    {
        return $this->currentStatus;
    }

    public function setCurrentStatus(?string $currentStatus): self
    {
        $this->currentStatus = $currentStatus;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: true)]
    private ?string $resumePath = null;

    public function getResumePath(): ?string
    {
        return $this->resumePath;
    }

    public function setResumePath(?string $resumePath): self
    {
        $this->resumePath = $resumePath;
        return $this;
    }

    #[ORM\Column(type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $lastUpdateDate = null;

    public function getLastUpdateDate(): ?\DateTimeInterface
    {
        return $this->lastUpdateDate;
    }

    public function setLastUpdateDate(?\DateTimeInterface $lastUpdateDate): self
    {
        $this->lastUpdateDate = $lastUpdateDate;
        return $this;
    }

    #[ORM\ManyToOne(targetEntity: User::class, inversedBy: 'applications')]
    #[ORM\JoinColumn(name: 'user_id', referencedColumnName: 'user_id')]
    private ?User $user = null;

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;
        return $this;
    }

    #[ORM\ManyToOne(targetEntity: Joboffer::class, inversedBy: 'applications')]
    #[ORM\JoinColumn(name: 'jobOfferId', referencedColumnName: 'jobOfferId')]
    private ?Joboffer $joboffer = null;

    public function getJoboffer(): ?Joboffer
    {
        return $this->joboffer;
    }

    public function setJoboffer(?Joboffer $joboffer): self
    {
        $this->joboffer = $joboffer;
        return $this;
    }

    #[ORM\Column(type: 'decimal', nullable: true)]
    private ?float $expectedSalary = null;

    public function getExpectedSalary(): ?float
    {
        return $this->expectedSalary;
    }

    public function setExpectedSalary(?float $expectedSalary): self
    {
        $this->expectedSalary = $expectedSalary;
        return $this;
    }

    #[ORM\Column(type: 'date', nullable: true)]
    private ?\DateTimeInterface $availabilityDate = null;

    public function getAvailabilityDate(): ?\DateTimeInterface
    {
        return $this->availabilityDate;
    }

    public function setAvailabilityDate(?\DateTimeInterface $availabilityDate): self
    {
        $this->availabilityDate = $availabilityDate;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: true)]
    private ?string $phone = null;

    public function getPhone(): ?string
    {
        return $this->phone;
    }

    public function setPhone(?string $phone): self
    {
        $this->phone = $phone;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: true)]
    private ?string $email = null;

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(?string $email): self
    {
        $this->email = $email;
        return $this;
    }

    #[ORM\Column(type: 'integer', nullable: true)]
    private ?int $experienceYears = null;

    public function getExperienceYears(): ?int
    {
        return $this->experienceYears;
    }

    public function setExperienceYears(?int $experienceYears): self
    {
        $this->experienceYears = $experienceYears;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: true)]
    private ?string $portfolioUrl = null;

    public function getPortfolioUrl(): ?string
    {
        return $this->portfolioUrl;
    }

    public function setPortfolioUrl(?string $portfolioUrl): self
    {
        $this->portfolioUrl = $portfolioUrl;
        return $this;
    }

    #[ORM\Column(type: 'decimal', nullable: true)]
    private ?float $score = null;

    public function getScore(): ?float
    {
        return $this->score;
    }

    public function setScore(?float $score): self
    {
        $this->score = $score;
        return $this;
    }

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $reviewNote = null;

    public function getReviewNote(): ?string
    {
        return $this->reviewNote;
    }

    public function setReviewNote(?string $reviewNote): self
    {
        $this->reviewNote = $reviewNote;
        return $this;
    }

}
